components/schemes/Typography/src/MDCTypographyScheme.m

```diff
@@ -84,8 +130,70 @@ - (id)copyWithZone:(NSZone *)zone {
   copy.caption = self.caption;
   copy.button = self.button;
   copy.overline = self.overline;
+  copy.mdc_adjustsFontForContentSizeCategory = self.mdc_adjustsFontForContentSizeCategory;
 
   return copy;
 }
 
+#pragma mark - Add scaling curved based on the font style
+
+- (void)addScalingCurves {
+  // Attach a sizing curve to all fonts
+  MDCFontScaler *fontScaler =
+      [[MDCFontScaler alloc] initForMaterialTextStyle:MDCTextStyleHeadline1];
+  _headline1 = [fontScaler scaledFontWithFont:_headline1];
+  _headline1 = [_headline1 mdc_scaledFontAtDefaultSize];
+
+  fontScaler = [[MDCFontScaler alloc] initForMaterialTextStyle:MDCTextStyleHeadline2];
+  _headline2 = [fontScaler scaledFontWithFont:_headline2];
+  _headline2 = [_headline2 mdc_scaledFontAtDefaultSize];
+
+  fontScaler = [[MDCFontScaler alloc] initForMaterialTextStyle:MDCTextStyleHeadline3];
+  _headline3 = [fontScaler scaledFontWithFont:_headline3];
+  _headline3 = [_headline3 mdc_scaledFontAtDefaultSize];
+
+  fontScaler = [[MDCFontScaler alloc] initForMaterialTextStyle:MDCTextStyleHeadline4];
+  _headline4 = [fontScaler scaledFontWithFont:_headline4];
+  _headline4 = [_headline4 mdc_scaledFontAtDefaultSize];
+
+  fontScaler = [[MDCFontScaler alloc] initForMaterialTextStyle:MDCTextStyleHeadline5];
+  _headline5 = [fontScaler scaledFontWithFont:_headline5];
+  _headline5 = [_headline5 mdc_scaledFontAtDefaultSize];
+
+  fontScaler = [[MDCFontScaler alloc] initForMaterialTextStyle:MDCTextStyleHeadline6];
+  _headline6 = [fontScaler scaledFontWithFont:_headline6];
+  _headline6 = [_headline6 mdc_scaledFontAtDefaultSize];
+
+  fontScaler = [[MDCFontScaler alloc] initForMaterialTextStyle:MDCTextStyleSubtitle1];
+  _subtitle1 = [fontScaler scaledFontWithFont:_subtitle1];
+  _subtitle1 = [_subtitle1 mdc_scaledFontAtDefaultSize];
+
+  fontScaler = [[MDCFontScaler alloc] initForMaterialTextStyle:MDCTextStyleSubtitle2];
+  _subtitle2 = [fontScaler scaledFontWithFont:_subtitle2];
+  _subtitle2 = [_subtitle2 mdc_scaledFontAtDefaultSize];
+
+  fontScaler = [[MDCFontScaler alloc] initForMaterialTextStyle:MDCTextStyleBody1];
+  _body1 = [fontScaler scaledFontWithFont:_body1];
+  _body1 = [_body1 mdc_scaledFontAtDefaultSize];
+
+  fontScaler = [[MDCFontScaler alloc] initForMaterialTextStyle:MDCTextStyleBody2];
+  _body2 = [fontScaler scaledFontWithFont:_body2];
+  _body2 = [_body2 mdc_scaledFontAtDefaultSize];
+
+  fontScaler = [[MDCFontScaler alloc] initForMaterialTextStyle:MDCTextStyleBody2];
```

Comment:
Is this a duplicated line? Copy-paste error?

Reply:
Almost certainly C&P.
Removed.